### PR TITLE
create command to add new member

### PR DESCRIPTION
**Pull request by RignonNoel**
| Q                                          | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                     | Features
| Tickets (_issues_) concerned               | None

---

##### What have you changed ?
Add command to create new user

##### How did you change it ?

 - Add new Django command
 - Add some tests
 - Add new function in `services.py` to send email

##### Is there a special consideration?
- No test on email domain in function of the university since it's a special case and we know the users.

Verification :
 -  [x] My branch name respect the standard defined in `CONTRIBUTING.md`
 -  [x] All my commits respect the standard defined in `CONTRIBUTING.md`
 -  [x] My additions are I18N
 -  [x] This Pull-Request fully meets the requirements defined in the issue
 -  [x] I added or modified the attached tests
 -  [x] I added or modified the documentation

### CODE REVIEW

File: blitz_api/management/commands/create_member.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            membership = Membership.objects.get(
+                pk=options['membership']
+            )
+            user.membership = membership
+            user.membership_end = datetime.datetime.now()
+            user.save()
+        except Membership.DoesNotExist:
+            # We make sure to delete the current user
+            # if we are not able to complete all the request
+            user.delete()
```
Comment (francisjeanneau) [inline blitz_api/management/commands/create_member.py]: Voir si c'est possible de faire une commande atomique plutôt

File: blitz_api/management/commands/create_member.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            is_active=True
+        )
+        user.set_password(options['password'])
+
+        try:
+            membership = Membership.objects.get(
+                pk=options['membership']
+            )
+            user.membership = membership
+            user.membership_end = datetime.datetime.now()
```
Comment (francisjeanneau) [inline blitz_api/management/commands/create_member.py]: C'est voulu de faire expirer le membership immédiatement?

File: blitz_api/services.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
 from django.conf import settings
 from django.core.mail import EmailMessage
 from django.utils.translation import ugettext_lazy as _
+from django.template.loader import render_to_string
 
 from .exceptions import MailServiceError
-
+from django.core.mail import send_mail as send_email
```
Comment (francisjeanneau) [inline blitz_api/services.py]: from django.core.mail import send_mail ~as send_email~

Comment (RignonNoel) [reply blitz_api/services.py]: Il y aura un conflit de nom avec la fonction dans le fichier si j'enleve le renommage. Je me suis dit qu'il etait preferable de faire un renommage le temps qu'on alloue du temps a la suppression de cette fonction deprecated

Comment (francisjeanneau) [reply blitz_api/services.py]: Ah oui j'avais pas vu. Ne devrais tu pas mieux renommer à `django_send_mail` plutôt que le même nom?

File: blitz_api/management/commands/create_member.py (hunk 1 above)
Comment (RignonNoel) [reply blitz_api/management/commands/create_member.py]: Merci pour le conseil!

File: blitz_api/management/commands/create_member.py (hunk 2 above)
Comment (RignonNoel) [reply blitz_api/management/commands/create_member.py]: Non, c'est un gros oubli! Bien vu!
J'ai ajoute une verification dans les tests.

File: blitz_api/services.py (hunk 3 above)
Comment (RignonNoel) [reply blitz_api/services.py]: :+1:

**Comment by RignonNoel** [comment]
@francisjeanneau We forgot multiple obligatory fields in this PR, i added them and some test to check, can you have a second check ? 

New special consideration:
 - No test on email domain in function of the university since it's a special case and we know the users.

**Comment by RignonNoel** [comment]
Also, test doesn't work when it come in a PR. Problem of envar, ca be good to let a default value to obligatory settings

**Comment by francisjeanneau** [comment]
> Also, test doesn't work when it come in a PR. Problem of envar, ca be good to let a default value to obligatory settings

This is desired for now, no configuration has been made for PR tests through Travis. If I am not mistaken, PRs do not have access to the encrypted environment variables.

**Comment by francisjeanneau** [comment]
Merged

**Pull request closed without merging**